general/arrays_by_name.hpp

```diff
@@ -0,0 +1,256 @@
+// Copyright (c) 2010-2023, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+#ifndef MFEM_ARRAYS_BY_NAME
+#define MFEM_ARRAYS_BY_NAME
+
+#include "../config/config.hpp"
+#include "array.hpp"
+
+#include <iostream>
+#include <map>
+#include <set>
+#include <string>
+
+namespace mfem
+{
+
+template <class T>
+class ArraysByName
+{
+protected:
+   /// Reusing STL map iterators
+   using container      = std::map<std::string,Array<T> >;
+   using iterator       = typename container::iterator;
+   using const_iterator = typename container::const_iterator;
+
+   /// Map containing the data sorted alphabetically by name
+   container data;
+
+public:
+
+   /// @brief Default constructor
+   ArraysByName() = default;
+
+   /// @brief Copy constructor: deep copy from @a src
+   ArraysByName(const ArraysByName &src) = default;
+
+   /// @brief Move constructor: deep move from @a src
+   ArraysByName(ArraysByName &&src) noexcept = default;
+
+   /// @brief Copy constructor (deep copy) from 'src', an ArraysByName
+   /// container of convertible type.
+   template <typename CT>
+   inline ArraysByName(const ArraysByName<CT> &src);
+
+   /// @brief Return the number of named arrays in the container
+   inline int Size() const { return data.size(); }
```

Comment:
```suggestion
   int Size() const { return data.size(); }
```
Don't need `inline` if the member function is defined in the class body (it is automatic).